packages/dds/merge-tree/src/test/mergeTree.annotate.deltaCallback.spec.ts

```diff
@@ -92,8 +92,8 @@ describe("MergeTree", () => {
             insertText(
                 mergeTree,
                 4,
-                remoteClientId,
                 remoteSequenceNumber,
+                remoteClientId,
```

Comment:
While writing this test I realized some of the above tests might not be doing exactly what they intend--the tests which involve overlapping annotate / remove operations end up applying the annotate on the range as viewed *after* the remove is applied, since there is logic in the segment length calculation which prevents any rebase behavior on local ops. So for example the test which does a remote delete of [4, 6) and then a local annotation of [3, 8) actually ends up annotating [3,8) of "hel world". I'm not sure if the tests are intending that or if they should be adjusted--either introducing a second remote client and treating the local client as an observer OR always applying remote ops after applying the local ops they're concurrent to would be two reasonable approaches.

Reply:
Looking at these tests they are not very useful. I think they are doing what they indented, it's just that what they intend is not useful. I agree the tests you call out would be more useful, and we do have coverage for may of those scenarios in merge tree, just not specifically around eventing.

Reply:
fair enough. I'll leave them be for now (though, still including the param ordering issue fix here :P)

Reply:
there are better tests in: packages\dds\merge-tree\src\test\client.applyMsg.spec.ts you could leverage that infra to cover your cases more easily